### PR TITLE
docs: touch up code edit experience

### PR DESCRIPTION
**Pull request by levithomason**
This PR makes some styling alignment tweaks to the new component examples and shorthand/children API selection experience.

# Before

image

# After

- align header / description / menu in examples
- minimize menu text
- differentiate rendered example with more padding
- dark theme editor
- move editor controls inside the editor (shorthand/children, copy/reset/edit on github)
- improve error experience
- improve all styling

### CODE REVIEW

File: docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx [hunk 1]
```diff
@@ -12,41 +13,37 @@ const ComponentControls: any = props => {
...
-      */}
-      <div>
-        <Menu color="green" icon="labeled" size="tiny" floated="right" text>
-          <ComponentControlsShowVariables active={showVariables} onClick={onShowVariables} />
-          <ComponentControlsRtl active={showRtl} onClick={onShowRtl} />
-          <ComponentControlsMaximize examplePath={examplePath} />
-          <ComponentControlsCopyLink anchorName={anchorName} onClick={onCopyLink} />
-        </Menu>
-      </div>
-    </Transition>
```
Comment (levithomason) [inline docs/src/components/ComponentDoc/ComponentControls/ComponentControls.tsx:37]: Transition was no longer used for example menus.

File: docs/src/components/ComponentDoc/ComponentControls/ComponentControlsCopyLink.tsx [hunk 2]
```diff
@@ -2,11 +2,9 @@ import PropTypes from 'prop-types'
 import React, { Component } from 'react'
 import { Icon, Menu } from 'semantic-ui-react'
 
-import ComponentControlsToolTip from './ComponentControlsToolTip'
-
```
Comment (levithomason) [inline docs/src/components/ComponentDoc/ComponentControls/ComponentControlsCopyLink.tsx:6]: Tooltips simply restated the menu text adding no value.  Removed this unnecessary distraction to the user.

File: docs/src/components/ComponentDoc/ComponentControls/ComponentControlsCopyLink.tsx [hunk 3]
```diff
@@ -2,11 +2,9 @@ import PropTypes from 'prop-types'
...
 export default class ComponentControlsCopyLink extends Component<any, any> {
   private mounted: boolean
-  private readonly btnLabel = 'Direct link'
+  private readonly btnLabel = 'Permalink'
```
Comment (levithomason) [inline docs/src/components/ComponentDoc/ComponentControls/ComponentControlsCopyLink.tsx:7]: Simplified example menu labels.

File: docs/src/components/ComponentDoc/ComponentControls/ComponentControlsShowCode.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    </Menu.Item>
+  )
+}
+
+ComponentControlsShowCode.propTypes = {
+  active: PropTypes.bool,
+  onClick: PropTypes.func,
+}
+
+export default updateForKeys(['active'])(ComponentControlsShowCode)
```
Comment (levithomason) [inline docs/src/components/ComponentDoc/ComponentControls/ComponentControlsShowCode.tsx:19]: Moved all example buttons back up to the example description area.  The multiple levels of buttons were too hard to follow.

File: docs/src/components/ComponentDoc/ComponentControls/ComponentControlsShowVariables.tsx [hunk 5]
```diff
@@ -3,18 +3,15 @@ import React from 'react'
 import { Icon, Menu } from 'semantic-ui-react'
 
 import { updateForKeys } from 'docs/src/hoc'
-import ComponentControlsToolTip from './ComponentControlsToolTip'
 
 const ComponentControlsShowVariables: any = ({ active, onClick }) => {
-  const btnLabel = 'Edit Variables'
+  const btnLabel = 'Theme it'
```
Comment (levithomason) [inline docs/src/components/ComponentDoc/ComponentControls/ComponentControlsShowVariables.tsx:8]: Change the "show code" and "show variables" to CTAs to make it more clear what they are for and invite the user to try them.

File: docs/src/components/Editor/Editor.tsx [hunk 6]
```diff
@@ -1,13 +1,12 @@
...
-import React, { CSSProperties, MouseEvent } from 'react'
+import React from 'react'
 import AceEditor, { AceEditorProps } from 'react-ace'
 import ace from 'brace'
 import 'brace/ext/language_tools'
 import 'brace/mode/jsx'
 import 'brace/mode/html'
-import 'brace/theme/tomorrow'
-import EditorPreview, { IEditorPreviewProps } from './EditorPreview'
+import 'brace/theme/tomorrow_night'
```
Comment (levithomason) [inline docs/src/components/Editor/Editor.tsx:9]: Dark theme 🎉

File: docs/src/components/Editor/Editor.tsx [hunk 7]
```diff
@@ -83,40 +80,29 @@ class Editor extends React.Component<IEditorProps> {
...
   constructor(props: IEditorProps) {
     super(props)
 
     this.setLineCount(props.value)
-    this.setShouldShowPreviewFlag(props)
   }
 
   public componentWillReceiveProps(nextProps: IEditorProps) {
     const previousPros = this.props
-    const { value, active, showCursor, readOnly } = nextProps
```
Comment (levithomason) [inline docs/src/components/Editor/Editor.tsx:102]: Read only styles were removed in favor of grayscale HTML, for now:

image

File: docs/src/components/ComponentDoc/ComponentControls/ComponentControlsShowCode.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { Icon, Menu } from 'semantic-ui-react'
+
+import { updateForKeys } from 'docs/src/hoc'
+
+const ComponentControlsShowCode: any = ({ active, onClick }) => {
+  const btnLabel = 'Try it'
```
Comment (dvdzkwsk) [inline docs/src/components/ComponentDoc/ComponentControls/ComponentControlsShowCode.tsx]: Not your doing, but just inline this guy? <3

File: docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx [hunk 9]
```diff
@@ -49,48 +49,19 @@ interface IComponentExampleState {
...
-    showCursor: false,
-    readOnly: true,
-    highlightActiveLine: false,
-    preview: {
-      size: 4,
-      label: 'Show more',
-      icon: 'html5',
-    },
-  } as IEditorProps,
-}
```
Comment (levithomason) [inline docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx:92]: These were used in only one location.  Inlined the props to reduce developer indirection.

File: docs/src/index.ejs [hunk 10]
```diff
@@ -38,7 +38,13 @@
       background: #f7f7f7;
     }
 
-    /* Anchors */
+    /* Ace editor overrides */
+
+    /*.ace-tomorrow-night {*/
+      /*background: #27282e;*/
```
Comment (dvdzkwsk) [inline docs/src/index.ejs]: Cruft?

File: docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx [hunk 11]
```diff
@@ -380,136 +357,129 @@ class ComponentExample extends PureComponent<IComponentExampleProps, IComponentE
...
-    // get component name from file path:
-    // elements/Button/Types/ButtonButtonExample
-    const pathParts = currentCodePath.split(__PATH_SEP__)
-    const filename = pathParts[pathParts.length - 1]
-
-    this.ghEditHref = [
-      `${repoURL}/edit/master/docs/src/examples/${currentCodePath}.tsx`,
-      `?message=docs(${filename}): your description`,
-    ].join('')
-  }
```
Comment (levithomason) [inline docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx:397]: This is a class mutation side-effect that was set in the "renderJSX" method but relied upon in the "renderCodeEditorMenu" method only.  I've inlined the calculation there instead of relying on outside mutations.

**Review by dvdzkwsk — DISMISSED** [review]
giphy

File: docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx [hunk 12]
```diff
@@ -380,136 +357,129 @@ class ComponentExample extends PureComponent<IComponentExampleProps, IComponentE
...
+    const { error } = this.state
+
+    if (!error) return null
+
+    return (
+      <Segment size="small" color="red" basic inverted padded secondary>
+        <pre>{error}</pre>
+      </Segment>
+    )
+  }
```
Comment (levithomason) [inline docs/src/components/ComponentDoc/ComponentExample/ComponentExample.tsx:478]: Separated error rendering from JSX rendering so we can show errors even after closing the editor.

# Open

image

# Closed

image

File: docs/src/components/Editor/EditorPreview.tsx [hunk 13]
```diff
@@ -1,78 +0,0 @@
...
-      </div>
-    )
-  }
-
-  private toPx(value: number): string {
-    return `${value}px`
-  }
-}
-
-export default EditorPreview
```
Comment (levithomason) [inline docs/src/components/Editor/EditorPreview.tsx:78]: There's no more preview editor needed.  We hide the code editor now.
